controllers/keda/hpa.go

```diff
@@ -145,9 +157,18 @@ func (r *ScaledObjectReconciler) updateHPAIfNeeded(ctx context.Context, logger l
 		return err
 	}
 
+	// check if hpa name is updated, and if so we need to delete the old hpa before creating new one
```

Comment:
TBH the implementation is quite confusing. We should have probably a separate check for this right in the reconcile loop. And have separate states:
1. Create a new HPA
2. Update existing HPA (no rename)
3. Rename HPA (create new and delete one)

Reply:
IMHO, the main logic should be here: https://github.com/kedacore/keda/blob/26edd4f551f8f2d6a4d3ddad862c77f757ebac84/controllers/keda/scaledobject_controller.go#L362

Reply:
Changed the flow. Now it's indeed seems better: https://github.com/kedacore/keda/pull/3068/commits/196e8f6bb76b9fe8487789e6b6b217eead474a5d